utils/vlog/vlog_to_pbtype.py

```diff
@@ -41,6 +41,14 @@
 
     - `(* PORT_CLASS="clock" *)` : specify the VPR "port_class"
 
+    - `(* FASM_xxxx *)` : All attributes with names starting from "FASM_" will
+        be converted do metadata entries with corresponding lowercase names.
+
+    - `(* FASM_xxxx_mmmm *)` : If there are modes and an attribute name begins
+       with "FASM" and ends with an upper case mode name, then it is applied
+       only for the matching mode. The base attribute name with mode name
+       removed is converted to lowercase "fasm_xxxx"
```

Comment:
We should just apply the `(* FASM_xxx *)` only inside the given mode using a `generate` statement.

Reply:
I am not sure if I understand your idea correctly. You mean apply `(* FASM_xxxx *)` only inside a generate statement? I don't think it is possible as attributes declared there apply not to the top level module but to child modules, nets etc.

As far as I understand we want to be able to specify different FASM features for the top level module for each mode, provided that it has modes. Right now it works just like that.

Could you see the `fasm_features_modes` test and tell if it is the correct behavior?